support/controlplane-component/defaults.go

```diff
@@ -31,26 +52,42 @@ var (
 	)
 )
 
-func (c *controlPlaneWorkload[T]) defaultOptions(cpContext ControlPlaneContext, podTemplateSpec *corev1.PodTemplateSpec, desiredReplicas *int32) (*config.DeploymentConfig, error) {
-	if _, exist := podTemplateSpec.Annotations[config.NeedMetricsServerAccessLabel]; exist || c.NeedsManagementKASAccess() ||
-		c.Name() == "packageserver" { // TODO: investigate why packageserver needs AutomountServiceAccountToken or set NeedsManagementKASAccess to true.
-		podTemplateSpec.Spec.AutomountServiceAccountToken = ptr.To(true)
+func (c *controlPlaneWorkload[T]) setDefaultOptions(cpContext ControlPlaneContext, workloadObj T, existingResources map[string]corev1.ResourceRequirements) error {
+	hcp := cpContext.HCP
+
+	labels := workloadObj.GetLabels()
+	if labels == nil {
+		labels = map[string]string{}
+	}
+	labels[managedByLabel] = "control-plane-operator-v2"
```

Comment:
I'd prefer if we don't change this since upper level tooling might be doing assumptions

Reply:
haven't thought about that, will change it back.